### ISSUE TITLE
Postgres migration from OpenSlides 2.3 to 3.x fails

### ISSUE THREAD

**Issue by jsaalfeld** [OpenSlides/OpenSlides#4266]
When migrating from OpenSlides 2.3 to the current OpenSlides version the migration of data from `sqlite` works fine, but it fails with migrations when `postgres` is chosen as a database-backend.

The sources tried had `postgres` running in version 9.6, since that is the version that was latest as 2.3 was often deployed.

It can be recreated with the data from the OpenSlides demo instance: [demo_data_2.3.zip](https://github.com/OpenSlides/OpenSlides/files/2835644/demo_data_2.3.zip)

The process is as followed (assuming database was named `openslides` and the database user is `openslides` as well) (ignore first step, if you are using the demo-data):

  1. export the database via `pg_dump -O -d openslides > dumpfile.sql` and copy it to the new system
  2. startup the new OpenSlides environment once and let migrations run to get an empty instance
  3. shut down OpenSlides, except for the `postgres` service
  4. drop the old `openslides` table with `dropdb openslides`
  5. create new database with `createdb -U openslides openslides`
  6. import the 2.3 dump with `psql -f dumpfile.sql -U openslides openslides`

The import will most likely generate some warnings for the grants, which should not matter. To finally migrate the data now, you can restart the run the migration process with `python manage.py migrate`. It will generate the following error:

```
Operations to perform:
  Apply all migrations: agenda, assignments, auth, contenttypes, core, mediafiles, motions, sessions, topics, users
Running migrations:
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
psycopg2.OperationalError: cannot ALTER TABLE "motions_motion" because it has pending trigger events


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage.py", line 11, in <module>
    exit(main())
  File "/app/openslides/__main__.py", line 41, in main
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 316, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 353, in execute
    output = self.handle(*args, **options)
  File "/app/openslides/core/management/commands/migrate.py", line 25, in handle
    super().handle(*args, **options)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 83, in wrapped
    res = handle_func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/commands/migrate.py", line 203, in handle
    fake_initial=fake_initial,
  File "/usr/local/lib/python3.7/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.7/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.7/site-packages/django/db/migrations/executor.py", line 244, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/local/lib/python3.7/site-packages/django/db/migrations/migration.py", line 124, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/usr/local/lib/python3.7/site-packages/django/db/migrations/operations/fields.py", line 150, in database_forwards
    schema_editor.remove_field(from_model, from_model._meta.get_field(self.name))
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/base/schema.py", line 469, in remove_field
    self.execute(self._delete_constraint_sql(self.sql_delete_fk, model, fk_name))
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/base/schema.py", line 133, in execute
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 100, in execute
    return super().execute(sql, params)
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 68, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 77, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.7/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.OperationalError: cannot ALTER TABLE "motions_motion" because it has pending trigger events

  Applying motions.0011_motion_version...
```

This will happen with `postgres` version 9.6 and 10. The interesting thing is, that you can directly alter the table `motions_motion` via the `psql` tool, but it doesn't work in the django-migrations. It seems to fail with the operation at [openslides/motions/migrations/0011_motion_version.py#L96](https://github.com/OpenSlides/OpenSlides/blob/eab6e9bfd0377b871edb786f7f28442c04e5ab48/openslides/motions/migrations/0011_motion_version.py#L96).

With `postgres` version 11 it will generate mutliple errors on the import, the migration itself will run fine though. The mess will only appear after starting OpenSlides and trying to login as the demo-User:

```
Internal Server Error: /apps/users/login/
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/base.py", line 126, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/base.py", line 124, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.7/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/rest_framework/views.py", line 495, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python3.7/site-packages/rest_framework/views.py", line 455, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python3.7/site-packages/rest_framework/views.py", line 492, in dispatch
    response = handler(request, *args, **kwargs)
  File "/app/openslides/users/views.py", line 479, in post
    return super().post(*args, **kwargs)
  File "/app/openslides/utils/views.py", line 30, in method_call
    return Response(self.get_context_data())
  File "/app/openslides/users/views.py", line 515, in get_context_data
    self.user.pk or 0, self.user.get_collection_string(), self.user.pk
  File "/usr/local/lib/python3.7/site-packages/asgiref/sync.py", line 64, in __call__
    return call_result.result()
  File "/usr/local/lib/python3.7/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/usr/local/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "/usr/local/lib/python3.7/site-packages/asgiref/sync.py", line 78, in main_wrap
    result = await self.awaitable(*args, **kwargs)
  File "/app/openslides/utils/cache.py", line 431, in get_element_restricted_data
    await self.update_restricted_data(user_id)
  File "/app/openslides/utils/cache.py", line 305, in update_restricted_data
    restricted_elements = await restricter(user_id, full_data)
  File "/app/openslides/utils/models.py", line 141, in restrict_elements
    return await cls.get_access_permissions().get_restricted_data(elements, user_id)
  File "/app/openslides/utils/access_permissions.py", line 54, in get_restricted_data
    return full_data if await self.async_check_permissions(user_id) else []
  File "/app/openslides/utils/access_permissions.py", line 38, in async_check_permissions
    return await async_has_perm(user_id, self.base_permission)
  File "/app/openslides/utils/auth.py", line 71, in async_has_perm
    raise RuntimeError(f"User with id {user_id} does not exist.")
RuntimeError: User with id 5 does not exist.
```

This error though will heavely depend on the dataset you are trying to import, sometimes the login still works, but something else will be broken. It seems, that the sequences in the postgres database are messed up big time, which, again, seems to correlate with the failure above, that some kind of trigger doesn't survive the migration.

**Comment by MaximilianKrambach** [comment]
Just a guess...

The migration `motions.0011_motion_version` seems to be the oldest of a series of migrations that do a `migrations.RunPython` involving a database field followed by some altering (mostly deleting) the field itself. Superficial internetsearching indicates that this can lead to problems.

**Comment by emanuelschuetze** [comment]
@ostcar Do you  have any idea for this issue?

**Comment by ostcar** [comment]
no. @normanjaeckel you are more experienced with migrations. (I hope we use a nosql database soon...)

**Comment by FinnStutzenstein** [comment]
I read something interesting in the django docs; I'll test this tomorrow.

**Comment by jsaalfeld** [comment]
fixed with #4272
